griptape/tasks/base_task.py

```diff
@@ -29,26 +29,32 @@ class State(Enum):
     parent_ids: list[str] = field(factory=list, kw_only=True)
     child_ids: list[str] = field(factory=list, kw_only=True)
     max_meta_memory_entries: Optional[int] = field(default=20, kw_only=True)
+    structure: Optional[Structure] = field(default=None, kw_only=True)
 
     output: Optional[BaseArtifact] = field(default=None, init=False)
-    structure: Optional[Structure] = field(default=None, init=False)
     context: dict[str, Any] = field(factory=dict, kw_only=True)
     futures_executor_fn: Callable[[], futures.Executor] = field(
         default=Factory(lambda: lambda: futures.ThreadPoolExecutor()),
         kw_only=True,
     )
 
+    def __attrs_post_init__(self) -> None:
+        if self.structure is not None and self not in self.structure.tasks:
+            self.structure.add_task(self)
+
     @property
     @abstractmethod
     def input(self) -> BaseArtifact: ...
 
     @property
     def parents(self) -> list[BaseTask]:
-        return [self.structure.find_task(parent_id) for parent_id in self.parent_ids]
+        return (
+            [self.structure.find_task(parent_id) for parent_id in self.parent_ids] if self.structure is not None else []
```

Comment:
I think it might be better to raise an exception if `self.structure` is `None` and this method property is invoked. The task may have parent_ids, but it just can't access them yet because it hasn't been added to the structure. As soon as its added to a structure then `len(task.parents)` will all of a sudden be greater than zero.